inst/include/dust/xoshiro.hpp

```diff
@@ -39,7 +47,7 @@ static inline uint64_t rotl(const uint64_t x, int k) {
 
 // This is the core generator (next() in the original C code)
 template <typename T>
-inline uint64_t xoshiro_next(rng_state_t<T>& state) {
+inline uint64_t xoshiro_next(rng_state_t<T> state) {
```

Comment:
Looking at the above I am less sure why this shouldn't be a reference. Did you say that there was a compiler error?
I guess it's a fairly minor copy anyway, as the object is small